xdsl/irdl/declarative_assembly_format.py

```diff
@@ -660,9 +660,11 @@ def parse_optional(self, parser: Parser, state: ParsingState) -> bool:
         return bool(operands)
 
     def parse_single_type(self, parser: Parser, state: ParsingState) -> None:
-        if len(state.operand_types) > 1:
-            parser.raise_error("Expected multiple types but received one.")
-        state.operand_types[0] = parser.parse_type()
+        pos_start = parser.pos
+        if s := self._set_using_variadic_index(
+            state.operand_types, "operand types", (parser.parse_type(),)
+        ):
+            parser.raise_error(s, at_position=pos_start, end_position=parser.pos)
```

Comment:
should these then be in a separate PR? With a separate test to demonstrate the impact of the change?

Reply:
Maybe, though I'm not quite sure how to test them, as the `functional-type` directive is the only thing that actually uses these functions

Reply:
I have an idea, will cook something up

Reply:
Now separated